src/main/java/com/es/phoneshop/web/ProductListPageServlet.java

```diff
@@ -13,29 +14,10 @@
 import java.util.List;
 
 public class ProductListPageServlet extends HttpServlet {
+    private ArrayListProductDao arrayListProductDao = new ArrayListProductDao();
```

Comment:
1) Field initialization
2) We must use interface ProductDao instead of ArrayListProductDao

private ProductDao productDao;

    @Override
    public void init() throws ServletException {
        productDao = new ArrayListProductDao();
    }